packages/rrweb/src/record/observer.ts

```diff
@@ -665,7 +665,9 @@ function initStyleSheetObserver(
             adds: [{ rule, index }],
           });
         }
-        return target.apply(thisArg, argumentsList);
+        return externalFunctionWrapper(() =>
```

Comment:
I only added this to insertRule and deleteRule since those are the ones that seem relevant for styled components. We can probably expand this further, but limiting the scope of the changes for now (though this should be fairly safe).

Reply:
Agreed could be useful for other areas that we would want to differentiate.